libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -449,6 +455,35 @@ constructor(
         tripSession.unregisterStateObserver(tripSessionStateObserver)
     }
 
+    /**
+     * Attach your own controller to determine when drivers arrived at stops via [ArrivalController]
+     * Use [navigateToNextStop] to manually move navigator to the next stop.
+     *
+     * @param arrivalController ArrivalObserver
+     */
+    fun attachArrivalController(arrivalController: ArrivalController) {
+        arrivalProgressObserver.attach(arrivalController)
+    }
+
+    /**
+     * Stop controlling arrival at stops, detaching will require the caller
+     * to call [navigateToNextStop] manually.
```

Comment:
Could we add a quick example in the test app showcasing this use case?

Reply:
hmm maybe need to add a manual mode to the MultipleStopsActivity?